File: hw/gudino/u3/hw29Singleton/SingletonPython/service/stock_notifier.py
from tkinter import messagebox

class StockNotifier:
    _instance = None

    def __init__(self):
        pass

    @classmethod
    def get_instance(cls):
        if cls._instance is None:
            cls._instance = StockNotifier()
        return cls._instance

    def alert_low_stock(self, product: str, stock: int):
        messagebox.showwarning(
            "Stock Alert",
            f"Low stock for product: {product}\nCurrent stock: {stock}"
        )

File: hw/gudino/u3/hw29Singleton/SingletonJava/src/main/java/service/StockNotifier.java
package service;

import javax.swing.JOptionPane;

public class StockNotifier {
    private static StockNotifier instance;

    private StockNotifier() {}

    public static StockNotifier getInstance() {
        if (instance == null) {
            instance = new StockNotifier();
        }
        return instance;
    }

    public void alertLowStock(String product, int stock) {
        JOptionPane.showMessageDialog(
            null,
            "Low stock for product: " + product + "\nCurrent stock: " + stock,
            "Stock Alert",
            JOptionPane.WARNING_MESSAGE
        );
    }
}
